src/catalog/database_catalog.cpp

```diff
@@ -1774,6 +1784,33 @@ Column DatabaseCatalog::MakeColumn(storage::ProjectedRow *const pr, const storag
   return col;
 }
 
+bool DatabaseCatalog::TryLock(transaction::TransactionContext *const txn) {
+  auto current_val = write_lock_.load();
+
+  const transaction::timestamp_t txn_id = txn->FinishTime();     // this is the uncommitted txn id
+  const transaction::timestamp_t start_time = txn->StartTime();  // this is the unchanging start time of the txn
+
+  const bool already_hold_lock = current_val == txn_id;
+  if (already_hold_lock) return true;
+
+  const bool owned_by_other_txn = (!transaction::TransactionUtil::Committed(current_val) && !already_hold_lock);
```

Comment:
nit: `!already_hold_lock` is always true because of the if-statement and return above.